scripts/test.sh

```
@@ -17,20 +19,23 @@ for module in "${modules[@]}"; do
-[ $VERIFY_EXIT -ne 0 ] && echo "Firebase test exited with error (verify)"
-exit_codes+=($VERIFY_EXIT)
-npx kill-port 8090 2>/dev/null || true
-cd .. || exit 1
+# Run modules that need Firebase emulator
+for module in "${firebase_modules[@]}"; do
+    echo "Running $module tests with Firebase emulator..."
+    cd "$module" || exit 1
+    npx firebase-tools --project="test" emulators:exec "go test -v -cover -coverprofile=\"$TEMP_DIR/$module.out\""
+    MODULE_EXIT=$?
+    [ $MODULE_EXIT -ne 0 ] && echo "Firebase test exited with error ($module)"
+    exit_codes+=($MODULE_EXIT)
```

Shellcheck (0.11.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)